signature-envelope-cose.md

```diff
@@ -0,0 +1,197 @@
+    'io.cncf.notary.signingScheme',
+    'io.cncf.notary.expiry'
+  ],
+  / cty / 3: 'application/vnd.cncf.notary.payload.v1+json',
+  'io.cncf.notary.signingScheme': 'notary.x509',
+  'io.cncf.notary.signingTime': 1234567890,
+  'io.cncf.notary.expiry': 1234567891
+}
+```
+
+Example with Signing Scheme `notary.x509.signingAuthority`
+
+```yaml
+{
+  / alg / 1: / PS384 / -38,
+  / crit / 2: [
+    'io.cncf.notary.signingScheme',
+    'io.cncf.notary.authenticSigningTime',
+    'io.cncf.notary.expiry'
+  ],
+  / cty / 3: 'application/vnd.cncf.notary.payload.v1+json',
+  'io.cncf.notary.signingScheme': 'notary.x509.signingAuthority',
+  'io.cncf.notary.authenticSigningTime': 1234567890,
+  'io.cncf.notary.expiry': 1234567891
+}
+```
+
+Note: The above examples are represented using the [extended CBOR diagnostic notation](https://datatracker.ietf.org/doc/html/rfc8152#appendix-C).
+
+- **[`alg`](https://datatracker.ietf.org/doc/html/rfc8152#section-3.1)** (*int*): This REQUIRED parameter (label `1`) defines which signing algorithm was used to generate the signature. The signature algorithm of the signing key (first certificate in `x5chain`) is the source of truth, and during signing the value of `alg` MUST be set corresponding to signature algorithm of the signing key using [this mapping](#supported-alg-header-values) that lists the Notary v2 allowed subset of `alg` values supported by COSE. Similarly verifier of the signature MUST match `alg` with signature algorithm of the signing key to mitigate algorithm substitution attacks.
+- **[`crit`](https://datatracker.ietf.org/doc/html/rfc8152#section-3.1)** (*array of int/tstr*): This REQUIRED parameter (label `2`) lists the header parameters that implementations MUST understand and process. It MUST only contain parameters apart from integer labels in the range of 0 to 8. This header MUST contain `io.cncf.notary.signingScheme` which is a required critical header, and optionally contain `io.cncf.notary.authenticSigningTime` and `io.cncf.notary.expiry` if these critical headers are present in the signature.
+- **[`content type`](https://datatracker.ietf.org/doc/html/rfc8152#section-3.1)** (*tstr*): The REQUIRED parameter content type (label `3`) is used to declare the media type of the secured content (the payload). The supported value is `application/vnd.cncf.notary.payload.v1+json`.
+- **`io.cncf.notary.signingScheme`** (*tstr*, critical): This REQUIRED header specifies the [Notary v2 Signing Scheme](./signing-scheme.md) used by the signature. Supported values are `notary.x509` and `notary.x509.signingAuthority`.
+- **`io.cncf.notary.signingTime`** (*date/time*): This header specifies the time at which the signature was generated. This is an untrusted date/time, and therefore not used in trust decisions. Its value is an Epoch-Based Date/Time defined in [RFC 8949](https://datatracker.ietf.org/doc/html/rfc8949#section-3.4.2). This claim is REQUIRED and only valid when signing scheme is `notary.x509`.
```

Comment:
@shizhMSFT  Is there a reason the RFC 3339 text based date time representation supported by COSE (tag 0) is not used in favor of epoch based representation (tag 1)? More concise representation may be one reason (~ 8 vs 20 bytes), are there other stronger reasons?

Reply:
COSE is based on CBOR where CBOR standards for **Concise Binary** Object Representation. Concise by default is the principle.

Oppositely, JWS is based on JSON, which is text-based not binary-based. That's why RFC3339 is preferable for JWS.